packages/replay/src/util/addEvent.ts

```diff
@@ -39,5 +41,10 @@ export async function addEvent(
     replay.getContext().earliestEvent = timestampInMs;
   }
 
-  return replay.eventBuffer.addEvent(event, isCheckout);
+  try {
+    return await replay.eventBuffer.addEvent(event, isCheckout);
+  } catch (error) {
+    __DEBUG_BUILD__ && logger.error(error);
+    replay.stop();
```

Comment:
Since this gets caught and doesn't bubble up, the callers will continue to run -- is that going to cause issues?

Reply:
It should be, I guess in the worst case we call `addEvent` again somewhere, which will no-op because it is stopped.